core/exceptions.py

```diff
@@ -12,6 +12,12 @@ class DotEnvException(PumpkinException):
     pass
 
 
+class ModuleDependencyException(PumpkinException):
+    """Raised when some module subclasses some other missing module."""
+
+    pass
```

Comment:
Why are you creating it if it doesn't do anything?

Reply:
Because than it can be used by modules other than core. The core modules will always be available, but that isn't true for other official repositories or even 3rd party ones.

Now, if module Bar depends on module Foo it can thow this exception to let the bot manager know that they haven't installed the Foo module, instead of throwing general `PumpkinException` or `ImportError`.

There may be other things in the `core/` that may not be directly used by the core or by base modules. But that doesn't mean that they shouldn't be there, because they will be used by some other module somewhere else.

Reply:
Understandable, then I don't like the `pass` being there. It might lead to some unexpected behavior. 

I would personally go the route of [This stackoverflow answer](https://stackoverflow.com/a/60465422) of the base having just a docstring and maybe modify the __init__ to support added arguments.

Reply:
The `pass` statement is [a null operation](https://docs.python.org/3/reference/simple_stmts.html#the-pass-statement), it can't cause any problems. By omitting everything and keeping just the docstring we're making it 100% compatible with its parent exception. [The Python 3 docs do this](https://docs.python.org/3/tutorial/errors.html#user-defined-exceptions).

We _may_ want to lock it a bit by requiring the module information, since it is module dependency exception. What should the error attribute be here? `<module name>`, `<repository name>.<module name>` or something else? Or we can make it more general `ModuleException`, which would hold module name and error message as arguments.

Reply:
I know the `pass` is in the documentation, it was more of the compatibility thing. The other part on the other hand applies for sure. 

If you have two exceptions, one based on Exception and the second one based on the first one, with both having just a docstring and a pass statement, that makes no sense. They should build up on each other.

We can actually make both `ModuleException` and `ModuleDependencyException`, with `ModuleDependencyException` being based on `ModuleException` with additional arguments/information/whatever. To properly describe the situation.